x-pack/plugins/cloud_security_posture/server/lib/check_index_status.ts

```diff
@@ -13,21 +13,42 @@ export const checkIndexStatus = async (
   esClient: ElasticsearchClient,
   index: string,
   logger: Logger,
-  postureType?: PostureTypes
+  postureType?: PostureTypes,
+  retentionTime?: number
 ): Promise<IndexStatus> => {
   const query =
     !postureType || postureType === 'all' || postureType === 'vuln_mgmt'
-      ? undefined
-      : {
+      ? {
           bool: {
-            filter: {
-              term: {
-                safe_posture_type: postureType,
+            must: {
+              range: {
+                '@timestamp': {
+                  gte: `now-${retentionTime}h/h`,
+                  lte: 'now/h',
+                },
               },
             },
           },
+        }
+      : {
+          bool: {
+            filter: [
+              {
+                term: {
+                  safe_posture_type: postureType,
```

Comment:
from the code I don't really understand why there is this difference between two queries, with or without `safe_posture_type: postureType`. It worth explaining in the comment at least

Reply:
I believe this is due to back then Prior to having CSPM feature, KSPM doesn't have a posture type field, as such when CSPM shows up it causes some issues with the findings. `safe_posture_type: postureType` in the query has always been there ever since